tests/test_mpi.py

```diff
@@ -945,6 +945,23 @@ def test_avoid_haloudate_if_flowdep_along_other_dim(self):
             assert np.allclose(f.data_ro_domain[0, 5:], [4.8, 4.8, 4.8, 4.8, 3.6], rtol=R)
             assert np.allclose(g.data_ro_domain[0, 5:], [4.8, 4.8, 4.8, 4.8, 2.], rtol=R)
 
+    @pytest.mark.parallel(mode=2)
+    def test_haloudate_scheduling_with_time_plus_no_time_varying_funcs(self):
+        grid = Grid(shape=(10,))
+        x = grid.dimensions[0]
+        t = grid.stepping_dim
+
+        f = Function(name='f', grid=grid, space_order=1)
+        g = TimeFunction(name='g', grid=grid, space_order=1, time_order=2)
+
+        eqns = [Eq(f, g[t, x+1] + g[t, x-1]),
+                Eq(g.forward, f[x-1] + f[x+1])]
+
+        op = Operator(eqns)
```

Comment:
I think we should also attempt execution and asserting against some expected values? since it's a fairly delicate test

Reply:
I've used `np.allclose` assertion as is the approach used in previous tests.